docs/design/datamover.md

```diff
@@ -0,0 +1,237 @@
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] User-facing documentation is created
+
+## Open questions
+
+* PVC/VolumeSnapshot mover - Should the Datamover Backup process be triggered off a PVC or a snapshot? 
+    * Should we support both types and provide user an option to pick either PVC or snapshot?
+
+
+## Summary
+OADP operator currently supports backup and restore of applications backed by CSI volumes by leveraging the Velero CSI plugin. The problem with CSI snapshots on some providers such as ODF is that these snapshots are local to the Openshift cluster and cannot be recovered if the cluster gets deleted accidentally or if there is a disaster. In order to overcome this issue, DataMover is made available for users to save the snapshots in a remote storage. 
+
+## Motivation
+
+Create an extensible design to support various data movers that can be integrated with OADP operator. Vendors should be able to bring their own data mover controller and implementation, and use that with OADP operator.
+
+## Goals
+* Create an extensible data mover solution
+* Supply a default data mover option 
+* Supply APIs for DataMover CRs (eg: DataMoverBackup, DataMoverRestore, DataMoverClass)
+* Supply a sample codebase for the Data Mover plugin and controller implementation
+
+
+## Non Goals
+* Maintain 3rd party data mover implementations
+* Adding a status watch controller to Velero
+
+## User stories
+
+Story 1: 
+As an application developer, I would like to save the CSI snaphots in a S3 bucket. 
+
+Story 2:
+As a cluster admin, I would like to be able to restore CSI snapshots if disaster happens.
+
+## Design & Implementation details
+
+This design supports adding the data mover feature to the OADP operator and facilitates integrating various vendor implemented data movers. 
+
+![DataMover CRD](../images/datamovercrd.png)
+
+Note: We will be supporting VolSync as the default data mover. 
+
+The DataMoverBackup Controller will watch for DataMoverBackup CR. Likewise, DataMoverRestore Controller will watch for DataMoverRestore CR. Both of these CRs will have a reference to a DataMoverClass. 
+
+`DataMoverClass` is a cluster scoped Custom Resource that will have details about the data mover provisioner. The provisioner (identified by the field `mover`) will be responsbile for creating a DataMover plugin that will create a `DataMoverBackup` CR & `DataMoverRestore` CR. They will also be responsible for implementing the logic for DataMoverBackup & DataMoverRestore controller that corresponds to their data mover. The spec will also include a field (`dmselector`) to identify the PVCs that would be moved with the given provisioner. 
+
+```
+apiVersion: oadp.openshift.io/v1alpha1
+kind: DataMoverClass
+metadata:
+  annotations:
+    oadp.openshift.io/default: "true"
+  name: <name>
+spec:
+  mover: <VolSync>
+  dmselector: <tagname>
```

Comment:
should this be pvcSelector or snapshotSelector?

since we are not sure, I am also open to keeping it just `selector` and differ the decision pvc vs snapshot